EIPS/eip-223.md

```diff
@@ -0,0 +1,385 @@
+```
+
+Get the total supply of the token. The functionality of the method is similar to `totalSupply()` of ERC-20. 
+
+#### `name`
+
+```solidity
+function name() view returns (string _name)
+```
+
+Get the name of the token. The functionality of the method is similar to `name()` of ERC-20.
+
+OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.
+
+#### `symbol`
+
+```solidity
+function symbol() view returns (bytes32 _symbol)
+```
+
+Get the symbol of token. The functionality of the method is similar to `symbol()` of ERC-20.
+
+OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.
+
+#### `decimals`
+
+```solidity
+function decimals() view returns (uint8 _decimals)
+```
+
+Get decimals of token. The functionality of the method is similar to `decimals()` of ERC-20.
+
+OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.
+
+#### `balanceOf`
+
+```solidity
+function balanceOf(address _owner) view returns (uint256 balance)
+```
+
+Get the account balance of another account with address `_owner`. The functionality of the method is similar to `balanceOf(address)` of ERC-20.
+
+
+#### `transfer(address, uint)`
+
+```solidity
+function transfer(address _to, uint _value) returns (bool)
+```
+
+This function must transfer tokens and invoke the function `tokenReceived(address, uint256, bytes calldata)` in `_to`, if `_to` is a contract. If the `tokenReceived` function is not implemented in `_to` (recipient contract), then the transaction must fail and the transfer of tokens must be reverted.
+If `_to` is an externally owned address, then the transaction must be sent without executing `tokenReceived` in `_to`.
+ `_data` can be attached to this token transaction and it will stay in blockchain forever (requires more gas). `_data` can be empty.
+
+#### `transfer(address, uint, bytes)`
+
+```solidity
+function transfer(address _to, uint _value, bytes calldata _data) returns (bool)
+```
+
+This function must transfer tokens and invoke the function `tokenReceived (address, uint256, bytes)` in `_to`, if `_to` is a contract. If the `tokenReceived` function is not implemented in `_to` (recipient contract), then the transaction must fail and the transfer of tokens must not occur. 
+If `_to` is an externally owned address, then the transaction must be sent without executing `tokenReceived` in `_to`.
+ `_data` can be attached to this token transaction and it will stay in blockchain forever (requires more gas). `_data` can be empty.
+
+NOTE: A possible way to check whether the `_to` is a contract or an address is to assemble the code of `_to`. If there is no code in `_to`, then this is an externally owned address, otherwise it's a contract.
+
+### Contract that is intended to receive [ERC-223](./eip-223.md) tokens
```

Comment:
I highly recommend this be at the bottom, below all the events, and that this be a solidity interface to indicate that this is completely separate from the above.